src/sentry/deletions/defaults/group.py

```diff
@@ -1,6 +1,59 @@
 from __future__ import absolute_import, print_function
 
-from ..base import ModelDeletionTask, ModelRelation
+from sentry import eventstore, nodestore
+from sentry.models import Event
+
+from ..base import BaseDeletionTask, BaseRelation, ModelDeletionTask, ModelRelation
+
+
+class GroupNodeDeletionTask(BaseDeletionTask):
+    """
+    Deletes nodestore data for group
+    """
+
+    DEFAULT_CHUNK_SIZE = 10000
+
+    def __init__(self, manager, group_id, project_id, **kwargs):
+        self.group_id = group_id
+        self.project_id = project_id
+        self.last_event = None
+        super(GroupNodeDeletionTask, self).__init__(manager, **kwargs)
+
+    def chunk(self):
+        conditions = []
+        if self.last_event is not None:
+            conditions.extend(
+                [
+                    ["timestamp", "<=", self.last_event.timestamp],
+                    [
+                        ["timestamp", "<", self.last_event.timestamp],
+                        ["event_id", "<", self.last_event.event_id],
+                    ],
+                ]
+            )
+
+        events = eventstore.get_events(
+            filter=eventstore.Filter(
+                conditions=conditions, project_ids=[self.project_id], group_ids=[self.group_id]
+            ),
+            limit=self.DEFAULT_CHUNK_SIZE,
+            referrer="deletions.group",
+            orderby=["-timestamp", "-event_id"],
+        )
+
+        if not events:
+            return False
+
+        self.last_event = events[-1]
+
+        node_ids = []
+        for event in events:
+            node_id = Event.generate_node_id(self.project_id, event.id)
+            node_ids.append(node_id)
+
+        nodestore.delete_multi(node_ids)
```

Comment:
What happens if this fails half way through and throws an exception?
Specifically:
- is delete_multi an atomic operation or can it fail with half the node deleted and half not?
- if this fails, how do we retry the deletion ?

Reply:
ok for delete-multi. 
Still this task can throw for other reasons. Do we retry in those cases ?

Reply:
Yeah, i think we run Celery tasks with infinite retry if any kind of exception is raised.